### PR TITLE
Ske v1.37 cherry picks

### PR DESCRIPTION
**Pull request by maboehm**
similar to v1.38 rebase (for provider-stackit). dropped:
- https://github.com/stackitcloud/gardener-extension-provider-openstack/commit/2d24b2b0ab38be5dd10b25b800c5d2ba4f497c0c terraform move blocks are no longer needed
- https://github.com/stackitcloud/gardener-extension-provider-openstack/commit/73ac4c40bac94c91d94ec3e18cc6ca43a878e065 old version bump
- https://github.com/stackitcloud/gardener-extension-provider-openstack/commit/2386623208fc50098e9556268c0fc9ec7a1e51e9 comment says this can be removed after 1.73
- https://github.com/stackitcloud/gardener-extension-provider-openstack/commit/f9a3b1f4273caccd5cf23dcd6cfadd73beb36027 vendor of older forked gardener version
- 7a228c4f this is in infra provider which is no longer used here.

### CODE REVIEW

**Comment by maboehm** [comment]
need to force push...

**Comment by maboehm** [comment]
I thought I could drop [6e81d16](https://github.com/stackitcloud/gardener-extension-provider-openstack/pull/48/commits/6e81d169b143cea0dcdfbcd3107ab8b615196fa4) and [ee95d6f](https://github.com/stackitcloud/gardener-extension-provider-openstack/pull/48/commits/ee95d6fead7da2551df47efa0701e0e2ea1e7319), but we are still using the types from our fork.
I guess the correct cleanup to do will be to reduce these changes / commits to only the API changes, which we can import in the provider-stackit.

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v1.37.1
+v1.37-ske-1
```
Comment (nschad) [inline VERSION:1]:
```suggestion
v1.37.1-ske-1
```

Wir haben die patch version eigentlich immer mitgenommen.

Comment (maboehm) [reply VERSION:1]: i cannot force push this :/ ...

**Comment by maboehm** [comment]
replaced by #49

**Pull request closed without merging**